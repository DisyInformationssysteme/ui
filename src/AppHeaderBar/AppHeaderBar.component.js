import React from 'react';
import { Link } from 'react-router';
import Dropdown from 'react-bootstrap/lib/Dropdown';
import MenuItem from 'react-bootstrap/lib/MenuItem';
import { api, Dispatcher, Icon } from 'react-cmf';
import { themr } from 'react-css-themr';
import { APP_HEADER_BAR } from '../identifiers';
// TODO: we have an issue during the build on this.


function getActions(context, id) {
	if (typeof id === 'string') {
		return api.action.getActionInfo(context, id);
	}
	return id;
}

/**
 * This is the main bar for webapp
 * @param {object} props   react props
 * @param {object} context react context
 */
function AppHeaderBar(props, context) {

	const actions = [];
	if (Array.isArray(props.actions)) {
		props.actions.forEach((action) => {
			actions.push(getActions(context, action));
		});
	}
	const userMenuActions = [];
	if (Array.isArray(props.userMenuActions)) {
		props.userMenuActions.forEach((action) => {
			userMenuActions.push(getActions(context, action));
		});
	}
	return (
		<div className={props.theme.appHeaderBar}>
			<div className={props.theme.brand}>
				<Link to="/" className={props.theme.link}>
					{props.logo ? <img
						className={props.theme.logo}
						src={props.logo.src}
						alt={props.logo.alt}
						style={props.logo.style}
					/> : null}
					{props.app}
				</Link>
			</div>
			<div className={props.theme.links}>
				{actions.map((action, i) => (
					<Dispatcher onClick={action} key={i}>
						<a className={props.theme.link}>
							<Icon name={action.icon} className="fa-fw" />
						</a>
					</Dispatcher>
				))}
				<Dropdown id="usermenu" pullRight>
					<Dropdown.Toggle useAnchor className="btn btn-inverse">
						<Icon name="icon-profile" className="fa-fw" />
						Micke TUCHEN
					</Dropdown.Toggle>
					<Dropdown.Menu >
						{userMenuActions.map((action, i) => (
							<Dispatcher onClick={action} key={i}>
								<MenuItem eventKey="1">
									<Icon name={action.icon} className="fa-fw" />
									{action.name}
								</MenuItem>
							</Dispatcher>
						))}
					</Dropdown.Menu>
				</Dropdown>
			</div>
		</div>
	);
}

AppHeaderBar.propTypes = {
	actions: React.PropTypes.arrayOf(React.PropTypes.string),
	userMenuActions: React.PropTypes.arrayOf(React.PropTypes.string),
	logo: React.PropTypes.shape({
		src: React.PropTypes.string,
		alt: React.PropTypes.string,
		style: React.PropTypes.string,
	}),
	app: React.PropTypes.string,
	theme: React.PropTypes.shape({
		appHeaderBar: React.PropTypes.string,
		brand: React.PropTypes.string,
		link: React.PropTypes.string,
		links: React.PropTypes.string,
		logo: React.PropTypes.string,
	}),
};
AppHeaderBar.contextTypes = {
	store: React.PropTypes.object,
};
export default themr(APP_HEADER_BAR, null)(AppHeaderBar);
export { AppHeaderBar };
