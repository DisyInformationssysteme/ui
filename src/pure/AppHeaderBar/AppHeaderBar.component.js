import React from 'react';
import { themr } from 'react-css-themr';
import classNames from 'classnames';
import { APP_HEADER_BAR } from '../../identifiers';

/**
 * The top bar is the place where the user finds useful information and tools
 * to describe the application
 * the logo, the name of the application, an optionnal global search,
 * few icons for feedback & onboarding and a menu to access logout
 * and profile information.
 * @param {object} props   react props
 */
function AppHeaderBar(props) {
	let logo = null;
	if (props.logo) {
		logo = (<img
			className={props.theme.logo}
			src={props.logo.src}
			alt={props.logo.alt}
			style={props.logo.style}
		/>);
	}
	let link = (
		<a href="#/" className={props.theme.link}>
			{logo} {props.app}
		</a>
	);
	if (props.link) {
		link = (
			<props.link>{logo} {props.app}</props.link>
		);
	}
	const wrapper = classNames(props.theme.appHeaderBar, 'navbar');
	const brand = classNames(props.theme.brand, 'navbar-brand');
	return (
		<div className={wrapper}>
			<div className={brand}>
				{link}
			</div>
			{props.children}
		</div>
	);
}

AppHeaderBar.propTypes = {
	logo: React.PropTypes.shape({
		src: React.PropTypes.string,
		alt: React.PropTypes.string,
		style: React.PropTypes.object,
	}),
	app: React.PropTypes.string,
	theme: React.PropTypes.shape({
		appHeaderBar: React.PropTypes.string,
		brand: React.PropTypes.string,
		link: React.PropTypes.string,
		logo: React.PropTypes.string,
	}),
};
export default themr(APP_HEADER_BAR)(AppHeaderBar);
export { AppHeaderBar as PureAppHeaderBar };
