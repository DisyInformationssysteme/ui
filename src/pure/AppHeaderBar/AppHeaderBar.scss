$tlnd-app-menu-bg-color: #465A65 !default;
$tlnd-app-menu-txt-color: white !default;

// AppHeaderBar styles
.appHeaderBar {
	background-color: $tlnd-app-menu-bg-color;
	height: 100%;
}
.brand {
	display: inline-block;
	width: 50%
}
.link, .link:hover {
	color: $tlnd-app-menu-txt-color;
	cursor: pointer;
	text-decoration: none;
	margin: 10px;
}

.links {
	display: inline-block;
	width: 50%;
	text-align: right;
	padding: 12px 20px 0;
}
.logo {
  max-width: '100px';
}
