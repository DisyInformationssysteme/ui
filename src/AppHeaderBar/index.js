import { themr } from 'react-css-themr';
import { APP_HEADER_BAR } from '../identifiers';
import AppHeaderBar from './AppHeaderBar.component';
import theme from './AppHeaderBar.scss';

const ThemedAppHeaderBar = themr(APP_HEADER_BAR, theme)(AppHeaderBar);

export default ThemedAppHeaderBar;
export { ThemedAppHeaderBar as AppHeaderBar };
