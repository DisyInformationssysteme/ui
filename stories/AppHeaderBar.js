import React from 'react';
import { storiesOf, action, linkTo } from '@kadira/storybook';

import { AppHeaderBar } from '../src/index';

storiesOf('App Header Bar', module)
	.add('default', () => {
		const logo = {
			src: 'https://www.talend.com/sites/all/themes/talend_responsive/images/talend-logo.png',
			alt: 'Talend logo',
			style: { maxWidth: '100px' },
		};
		return (
			<div>
				<h1>AppHeaderBar</h1>
				<h2>Definition</h2>
				<a href="http://guidelines.talend.com/document/92132#/ui-controls/app-header">Spec</a>
				<h2>Examples</h2>
				<AppHeaderBar
					logo={logo}
					app="Example app name"
				/>
			</div>
		);
	}
);
