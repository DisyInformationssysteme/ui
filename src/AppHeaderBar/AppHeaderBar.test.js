import React from 'react';
import renderer from 'react-test-renderer';
import AppHeaderBar from './AppHeaderBar.component';

jest.mock('react-dom');

describe('AppHeaderBar', () => {
	it('should render its name', () => {
		const appheaderbar = renderer.create(
			<AppHeaderBar name="Hello world" />
		).toJSON();
		expect(appheaderbar).toMatchSnapshot();
	});
});
