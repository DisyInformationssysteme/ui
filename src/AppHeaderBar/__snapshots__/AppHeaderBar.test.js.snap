exports[`AppHeaderBar should render its name 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <a
      className={undefined}
      onClick={[Function bound handleClick]}
      style={Object {}} />
  </div>
  <div
    className={undefined}>
    <div
      className="dropdown btn-group">
      <a
        aria-expanded={false}
        aria-haspopup={true}
        className="btn btn-inverse dropdown-toggle"
        href=""
        id="usermenu"
        onClick={[Function bound handleClick]}
        onKeyDown={[Function bound handleKeyDown]}
        role="button">
        <i
          className="fa fa-fw icon-profile" />
        Micke TUCHEN
         
        <span
          className="caret" />
      </a>
      <ul
        aria-labelledby="usermenu"
        className="dropdown-menu dropdown-menu-right"
        role="menu" />
    </div>
  </div>
</div>
`;
